Shared/Samples/Augment reality to navigate route/AugmentRealityToNavigateRouteView.swift

```diff
@@ -0,0 +1,253 @@
+// limitations under the License.
+
+import ArcGIS
+import ArcGISToolkit
+import AVFoundation
+import SwiftUI
+
+struct AugmentRealityToNavigateRouteView: View {
+    /// The data model for the selected route.
+    @StateObject private var routeDataModel = RouteDataModel()
+    /// A Boolean value indicating whether the route planner is showing.
+    @State private var isShowingRoutePlanner = true
+    /// The location datasource that is used to access the device location.
+    @State private var locationDataSource = SystemLocationDataSource()
+    /// A scene with an imagery basemap.
+    @State private var scene = Scene(basemapStyle: .arcGISImagery)
+    /// The elevation surface set to the base surface of the scene.
+    @State private var elevationSurface: Surface = {
+        let elevationSurface = Surface()
+        elevationSurface.navigationConstraint = .unconstrained
+        elevationSurface.opacity = 0.5
+        elevationSurface.backgroundGrid.isVisible = false
+        return elevationSurface
+    }()
+    /// The elevation source with elevation service URL.
+    @State private var elevationSource = ArcGISTiledElevationSource(url: URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!)
+    /// The graphics overlay containing a graphic.
+    @State private var graphicsOverlay = GraphicsOverlay()
+    /// The status text displayed to the user.
+    @State private var statusText = "Adjust calibration before starting."
+    /// A Boolean value indicating whether the use is navigatig the route.
+    @State private var isNavigating = false
+    /// The result of the route selected in the route planner view.
+    @State private var routeResult: RouteResult?
+    
+    init() {
+        elevationSurface.addElevationSource(elevationSource)
+        scene.baseSurface = elevationSurface
+    }
+    
+    var body: some View {
+        if isShowingRoutePlanner {
+            RoutePlannerView(isShowing: $isShowingRoutePlanner)
+                .onDidSelectRoute { routeGraphic, routeResult  in
+                    self.routeResult = routeResult
+                    graphicsOverlay = makeRouteOverlay(
+                        routeResult: routeResult,
+                        routeGraphic: routeGraphic
+                    )
+                }
+                .task {
+                    Task { try await elevationSource.load() }
+                }
+        } else {
+            WorldScaleSceneView { _ in
+                SceneView(scene: scene, graphicsOverlays: [graphicsOverlay])
+            }
+            .calibrationButtonAlignment(.bottomLeading)
+            .task {
+                Task {
+                    try await locationDataSource.start()
+                    
+                    for await location in locationDataSource.locations {
+                        try await routeDataModel.routeTracker?.track(location)
+                    }
+                }
```

Comment:
`Task` shouldn't be used just to discard an error.

```suggestion
                try? await locationDataSource.start()
                
                for await location in locationDataSource.locations {
                    try? await routeDataModel.routeTracker?.track(location)
                }
```